main.tf

```
@@ -21,6 +21,13 @@ data "aws_subnet" "private_subnets" {
   id    = var.private_subnet_ids[count.index]
 }
 
+locals {
+  node_group_subnets = length(var.node_pool_azs) != 0 ? [
+    for index, subnet in data.aws_subnet.private_subnets : subnet if contains(var.node_pool_azs, subnet.availability_zone)
+  ] : data.aws_subnet.private_subnets
```

```suggestion
  node_group_subnets = var.enable_nodes_use_public_subnet ? length(var.node_pool_azs) != 0 ? [
    for index, subnet in data.aws_subnet.public_subnets : subnet if contains(var.node_pool_azs, subnet.availability_zone) 
  ] : data.aws_subnet.public_subnets : length(var.node_pool_azs) != 0 ? [
    for index, subnet in data.aws_subnet.private_subnets : subnet if contains(var.node_pool_azs, subnet.availability_zone)
  ] : data.aws_subnet.private_subnets
```


And let's move these locals to the current locals block: https://github.com/streamnative/terraform-aws-cloud/pull/133/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR48